### PR TITLE
Add @WorkflowInit docs for Java

### PR DESCRIPTION
**Pull request by Quinn-With-Two-Ns**
## What does this PR do?
Add `@WorkflowInit` docs for Java based on the Python docs for the same feature https://github.com/temporalio/documentation/pull/3108

## Notes to reviewers

Note sure if we need to callout the SDK version this feature is available in? We didn't in Python, but maybe we should have?

### CODE REVIEW

File: docs/develop/java/message-passing.mdx [hunk 1]
```diff
@@ -508,6 +508,57 @@ public interface MyWorkflow {
 
 See [Finishing handlers before the Workflow completes](/encyclopedia/workflow-message-passing#finishing-message-handlers) for more information.
 
+### Use `@WorkflowInit` to operate on Workflow input before any handler executes
+
+Normally, your Workflows constructor won't have any parameters.
+However, if you use the `@WorkflowInit` annotation on your constructor, you can give it the same [Workflow parameters](/develop/java/core-application#workflow-parameters) as your `@WorkflowMethod`.
+The SDK will then ensure that your constructor receives the Workflow input arguments that the [Client sent](/develop/python/temporal-clients#start-workflow-execution).
+(The Workflow input arguments are also passed to your `@WorkflowMethod` method -- that always happens, whether or not you use the `@WorkflowInit` annotation.)
+This is useful if you have message handlers that need access to workflow input: see [Initializing the Workflow first](/encyclopedia/workflow-message-passing#workflow-initializers).
```
Comment (fairlydurable) [inline docs/develop/java/message-passing.mdx:517]:
```suggestion
This is useful if you have message handlers that need access to Workflow input: see [Initializing the Workflow first](/encyclopedia/workflow-message-passing#workflow-initializers).
```

File: docs/develop/java/message-passing.mdx [hunk 2]
```diff
@@ -508,6 +508,57 @@ public interface MyWorkflow {
 
 See [Finishing handlers before the Workflow completes](/encyclopedia/workflow-message-passing#finishing-message-handlers) for more information.
 
+### Use `@WorkflowInit` to operate on Workflow input before any handler executes
+
+Normally, your Workflows constructor won't have any parameters.
+However, if you use the `@WorkflowInit` annotation on your constructor, you can give it the same [Workflow parameters](/develop/java/core-application#workflow-parameters) as your `@WorkflowMethod`.
+The SDK will then ensure that your constructor receives the Workflow input arguments that the [Client sent](/develop/python/temporal-clients#start-workflow-execution).
+(The Workflow input arguments are also passed to your `@WorkflowMethod` method -- that always happens, whether or not you use the `@WorkflowInit` annotation.)
```
Comment (fairlydurable) [inline docs/develop/java/message-passing.mdx:516]:
```suggestion
The Workflow input arguments are also passed to your `@WorkflowMethod` method -- that always happens, whether or not you use the `@WorkflowInit` annotation.
```

File: docs/develop/java/message-passing.mdx [hunk 3]
```diff
@@ -508,6 +508,57 @@ public interface MyWorkflow {
 
 See [Finishing handlers before the Workflow completes](/encyclopedia/workflow-message-passing#finishing-message-handlers) for more information.
 
+### Use `@WorkflowInit` to operate on Workflow input before any handler executes
+
+Normally, your Workflows constructor won't have any parameters.
+However, if you use the `@WorkflowInit` annotation on your constructor, you can give it the same [Workflow parameters](/develop/java/core-application#workflow-parameters) as your `@WorkflowMethod`.
+The SDK will then ensure that your constructor receives the Workflow input arguments that the [Client sent](/develop/python/temporal-clients#start-workflow-execution).
+The Workflow input arguments are also passed to your `@WorkflowMethod` method -- that always happens, whether or not you use the `@WorkflowInit` annotation.
+This is useful if you have message handlers that need access to Workflow input: see [Initializing the Workflow first](/encyclopedia/workflow-message-passing#workflow-initializers).
```
Comment (fairlydurable) [inline docs/develop/java/message-passing.mdx:517]:
```suggestion
This is useful if you have message handlers that need access to Workflow input: see [Initializing the Workflow first](/encyclopedia/workflow-message-passing#workflow-initializers.
```

**Comment by fairlydurable** [comment]
Removed parentheses. Uppercased Workflow. Will align with .NET in future ticket.

**Review by fairlydurable — APPROVED** [review]
Unblocking.

Will merge when @drewhoskins-temporal gives okay

File: docs/develop/java/message-passing.mdx [hunk 4]
```diff
@@ -506,6 +506,57 @@ public interface MyWorkflow {
 
 See [Finishing handlers before the Workflow completes](/encyclopedia/workflow-message-passing#finishing-message-handlers) for more information.
 
+### Use `@WorkflowInit` to operate on Workflow input before any handler executes
+
+Normally, your Workflows constructor won't have any parameters.
+However, if you use the `@WorkflowInit` annotation on your constructor, you can give it the same [Workflow parameters](/develop/java/core-application#workflow-parameters) as your `@WorkflowMethod`.
+The SDK will then ensure that your constructor receives the Workflow input arguments that the [Client sent](/develop/python/temporal-clients#start-workflow-execution).
+The Workflow input arguments are also passed to your `@WorkflowMethod` method -- that always happens, whether or not you use the `@WorkflowInit` annotation.
+This is useful if you have message handlers that need access to Workflow input: see [Initializing the Workflow first](/encyclopedia/workflow-message-passing#workflow-initializers.
```
Comment (drewhoskins-temporal) [inline docs/develop/java/message-passing.mdx:515]: this is missing a paren.  @Quinn-With-Two-Ns @fairlydurable  Did we test this?

Comment (fairlydurable) [inline docs/develop/java/message-passing.mdx:515]:
```suggestion
This is useful if you have message handlers that need access to Workflow input: see [Initializing the Workflow first](/encyclopedia/workflow-message-passing#workflow-initializers).
```

Comment (fairlydurable) [reply docs/develop/java/message-passing.mdx]: Should be fixed now!

**Comment by fairlydurable** [comment]
Parenthesis returned to link.

**Comment by fairlydurable** [comment]
Thank you for putting this together @Quinn-With-Two-Ns and @drewhoskins-temporal

**Pull request merged**